Update the documentation in the combined repo for initial release

whowutwut opened the pull request:
This PR is not intended to refactor all the README files, but is intended to at least get the documentation to a good base so that we can release and the READMEs have continuity with  the  combined repo and correct links.

There's two use cases here that I feel we should keep the  documentation separate: 
* Customers (IBM Knowledge Center  (official) + RTD tracking it)
* Developers that may want to contribute to our project

- [x] Create a new README.md landing page for the combined repo, 
    * Keep it clean 
- [x] Remove the references to the planned deprecated repos (ibm-spectrum-scale-csi-driver, ibm-spectrum-scale-csi-operator) from the various saved README.[operator|driver].md files 
- [x] Enable RTD for the new combined repo in order to keep documentation living in-sync with code
* ~~double check the issue templates~~  Will open another issue for this


If anyone wants to co-contribute changes to  the PR branch, msg me to write  access to my repo.

whowutwut commented:
FYI, you can see the output of the README file by looking at my forked copy. 
https://github.com/whowutwut/ibm-spectrum-scale-csi/blob/doc1/README.md

The RTD is building to here: https://ibm-spectrum-scale-csi.readthedocs.io/en/latest/
But that  won't get rendered until the PR is merged or you decide to build it yourself locally.   If there's no objections, I'd like to suggest that we merge this first and make smaller changes as needed.

whowutwut commented:
This PR is ready for review.  I am not planning on adding more changes unless there are review comments.

mew2057 commented on docs/source/conf.py:
```diff
@@ -17,7 +17,7 @@
-project = 'IBM Spectrum Scale CSI Operator'
+project = 'IBM Spectrum Scale CSI'
 copyright = '2019, IBM'
 author = 'John Dunham'
```
Eventually we might want to change this author  haha.

whowutwut replied:
Can we just leave it blank?

mew2057 commented on docs/source/developers/deployment.rst:
```diff
@@ -5,13 +5,13 @@ Deployment
 --------------------
 
-In order to use the container image that you just built in the previous step, the image needs to be pushed to some container repository.
+In order to more easily consume the container image that you just built, the images needs to be pushed to some container repository.
```
The `container image` sounds awkward, but I think it's valid. Just noting  a potential confusion point.

whowutwut replied:
I will try and reword

smitaraut commented on docs/source/developers/deployment.rst:
```diff
@@ -29,74 +29,81 @@ Once you have a repository ready:
-  # Tag the build 
-  docker tag ibm-spectrum-scale-csi-operator quay.io/<your-user>/ibm-spectrum-scale-csi-operator:v0.9.1
+  # Configure some variables
+  VERSION="v0.9.1"
```
We should change to latest available version number 1.0.0

whowutwut replied:
We could.. 

The intent here is not to track the current version of the  code (because that's a maintainability nightmare) But since this is building the code locally and tagging it, would having this variable at 0.0.1 be better to disconnect the manual build with our product build version ?

smitaraut commented on docs/source/developers/deployment.rst:
```diff
@@ -29,74 +29,81 @@ Once you have a repository ready:
+  kubectl apply -f ${OPERATOR_DIR}/deploy/role.yaml
+  kubectl apply -f ${OPERATOR_DIR}/deploy/role_binding.yaml
+  kubectl apply -f ${OPERATOR_DIR}/deploy/service_account.yaml
+  kubectl apply -f ${OPERATOR_DIR}/deploy/crds/ibm-spectrum-scale-csi-operator-crd.yaml
```
Should we instruct the use of autogenerated file ibm-spectrum-scale-csi-operator.yaml ?

whowutwut replied:
Sure, let me make a change for this.

whowutwut replied:
On 2nd thought, this section is targeted for developers that may want contribute changes to the operator.... The changes to these yaml need to happen in those separate files, and the  generated file is created  from that.  

So asking the user to apply/delete that generated  file may cause confusion and would also fail the CI/CD if that is checked in, without the other files checked in.

smitaraut commented:
Pre-reqs for building the code should be mentioned - ansible, go, python (v2 or 3), pip/pip3, operator-sdk, docker, any specific version requirements

whowutwut commented:
Addressed some of the review comments above in subsequent commits: 

> Pre-reqs for building the code should be mentioned - ansible, go, python (v2 or 3), pip/pip3, operator-sdk, docker, any specific version requirements

I feel that the pre-reqs should be handled by the various playbooks that we provide as helper.. and with use, we would solidify the helper script.  

Lots of things may change in the upcoming months and it's better to have the code resolve the requirements if they really are HARD requirements, than to continue to maintain the documentation in real time, for now.

mew2057 commented:
\lgtm

whowutwut commented:
Open to restart  CI
